### ISSUE TITLE
Docker Builds for linux/arm, linux/arm64

### ISSUE THREAD

**Issue by rehashedsalt** [netbox-community/netbox-docker#468]
## Desired Behavior

Container images on Dockerhub and Quay should include builds for the following architectures in addition to `linux/amd64`:
* `linux/arm`
* `linux/arm64`

## Contrast to Current Behavior

Current container images only support `linux/amd64`.

## Changes Required

Presumably just build scripts, but any automated testing might need to be adjusted as well. I was able to build an image just fine by adding this to `build.sh` (with HEAD at `cb5ffa0`, for the context of the diff):

```diff
diff --git a/build.sh b/build.sh
index 82084bf..65a896c 100755
--- a/build.sh
+++ b/build.sh
@@ -352,6 +352,7 @@ for DOCKER_TARGET in "${DOCKER_TARGETS[@]}"; do
     if [ -n "${NO_PROXY}" ]; then
       DOCKER_BUILD_ARGS+=(--build-arg "no_proxy=${NO_PROXY}")
     fi
+    DOCKER_BUILD_ARGS+=(--platform "linux/arm,linux/arm64,linux/amd64")
 
     ###
     # Building the docker image
```

I'd PR this in, but this lame edit makes a ton of assumptions about your build system.

## Discussion: Benefits and Drawbacks

The biggest gain from adding these images is direct support for deployment to smaller machines like the Raspberry Pi. I, like a growing number of people, like to use a cluster of Pis as a cheap but effective lab environment. It'd be very nice to use the upstream Netbox image in my k8s cluster instead of having to rely on rebuilds.

It's not a terribly new change, either: the `linuxserver/netbox` image has supported these architectures for some time now.

Naturally, drawbacks will include widened support, but the odds of another couple architectures affecting this repo in particular are basically nil. If anything, it would give ARM-specific bugs in Netbox more exposure.

**Comment by cimnine** [comment]
Did you accomplish building Netbox for _arm_? I tried on the past weekend, see my personal branch (see https://github.com/netbox-community/netbox-docker/compare/develop...cimnine:Buildx). But the cryptography module did not build on my machine, at least not using the QEMU environment.

**Comment by ryanmerolle** [comment]
I say close this issue out unless OP wants to fix the PR and potentially support this on arm long term.

**Comment by anthr76** [comment]
Has any traction been made on this. I'd prefer using an upstream image. What specific help is needed from the community?

**Comment by anubisg1** [comment]
i would also like to see this.
How can i help?

**Comment by ryanmerolle** [comment]
> i would also like to see this.
> How can i help?

I cannot see the ci logs for this PR anymore, but both this PR and #604 just need to be sorted out with a useable image.

**Comment by cimnine** [comment]
> How can i help?

If you like, you could take #664 as a start, as it mostly lacks the changes to the automated build (the GitHub Actions) to become mergable.

#604 does way more, possibly even too much. It's a major refactoring. But we currently lack the resources to finish it. Also it seems to be blocked by some PostgreSQL Client problem.

**Comment by anubisg1** [comment]
> s it mostly lacks the changes to the automated build (the GitHub Actions) to become mergable.
> 
> #604 does way more, possibly even too much. It's a major refactor

ok, let me look into it. 

all logs are gone, but hopefully i can get something done anyway

**Comment by ryanmerolle** [comment]
For inspiration, nautobot just added support as well.  https://github.com/nautobot/nautobot/pull/1219

**Comment by mblaschke-daimlertruck** [comment]
Any updates on this?
ARM64 docker image is needed for Apple MacBooks with M1 and for cloud VMs with ARM64 CPU.

**Comment by ryanmerolle** [comment]
> ARM64 docker image is needed for Apple MacBooks with M1 and for cloud VMs with ARM64 CPU.

@mblaschke-daimlertruck anyone interested in this feature is likely using an M1 or M2 Apple MacOS device.  No need to remind us.

#664 is working on this and the docker image `tgenannt/netbox` is built for arm to test aginst.

**Comment by tobiasge** [comment]
ARM64 images are being build from my repository [here](https://github.com/tobiasge/netbox-docker) and published to [docker.io](https://hub.docker.com/r/tgenannt/netbox/tags) and [quay.io](https://quay.io/repository/tgenannt/netbox?tab=tags).

If the builds are stable we will continue with merging into the main repo.

**Comment by tobiasge** [comment]
Closed with #797
